ZERO-1693: implement FlowValve and SwitchValve and document AI instructions (#297)
foundation-zero/zero: zero-ui/docs/mimics/authoring-workflow.md

Reviewer: This whole document is rather verbose and wordy (canonical, non-negotiable etc). I think that gets in the way of the actual message. Could you condense it?

Author: I will add an introduction to this document to clarify that this is an instruction document for {insert favourite AI}. It might be rather verbose but it is not really meant for us humans. I asked CoPilot to write this for himself so that we always get more or less the same sort of output when asked to extract a component from Figma.

# Mimic Component Authoring Workflow

> This document is the canonical process for building new mimic components from Figma into Zero UI.

Reviewer: ```suggestion
This document how to build mimic components from Figma into Zero UI.
```

> This document is the canonical process for building new mimic components from Figma into Zero UI.

Reviewer: Is there a workflow for updating the component based on changes in Figma?

Author: Not right now, but I will add this when the opportunity presents itself.

Reviewer: I was under the impression that that was part of the challenge to solve: doing a one-time import is simple enough. Handling updates is where things get tricky

Author: I've added a separate instructions file for maintenance and tested this with adding the three way valve to the Valve component. While this still required some refactoring, the overall process was successful. I expect simple design changes to be fully automated.

> Use this workflow for every new mimic component to keep implementation quality and consistency high.

Reviewer: Less is more? If it's documented, it's the way it should be used. No need to state what should be obvious

Author: I'm leaning towards keeping it in, since CoPilot added this for itself. I find that treating it like a child and keep repeating yourself yields better results.

## Scope

Use this for domain-specific mimic components in:

- `src/modules/thrapp/mimics/components/**`

These components should be small, composable primitives intended for larger mimic scenes.

## Non-Negotiable Rules

- **Pure SVG template only** for all components.
- **No HTML wrappers** (`div`, `span`, etc.) around components.
- **One base geometry** when states are rotational variants.
- **Semantic tokens only** for color (no hardcoded hex/rgb in final component).
- **Typed state prop** with explicit union types.

> - **Exact Figma geometry** (exported vectors), avoid hand-drawn approximations.

Reviewer: ```suggestion
- **Exact Figma geometry** (exported vectors), do not use hand-drawn approximations.
```

- **Documentation required** for every component with examples and token notes.
- **Zero diagnostics** on all touched files before merge.
- **All text labels as SVG text** unless exact glyph shapes are required, then use path data from Figma.
- **Created files must be formatted and linted** according to project standards.

> - **State to colour mapping should be done by a computed property using a Record<Type, string>** for maintainability and clarity.

Reviewer: I thought we were en-US (colour)

Author: CoPilot is, but this was my own addition and I sometimes tend towards en-GB 😅.
